lib/src/facts/windows/ssh_resolver.cc

```diff
@@ -0,0 +1,122 @@
+#include <internal/facts/windows/ssh_resolver.hpp>
+#include <leatherman/file_util/file.hpp>
+#include <facter/util/string.hpp>
+
+#include <leatherman/util/environment.hpp>
+#include <leatherman/windows/file_util.hpp>
+#include <leatherman/windows/system_error.hpp>
+#include <leatherman/windows/user.hpp>
+#include <leatherman/windows/windows.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <boost/algorithm/string.hpp>
+#include <leatherman/logging/logging.hpp>
+#include <boost/algorithm/string.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/format.hpp>
+#include <tuple>
+#include <vector>
+
+#ifdef USE_OPENSSL
+#include <internal/util/scoped_bio.hpp>
+#include <openssl/sha.h>
+#include <openssl/evp.h>
+using namespace facter::util;
+#endif  // USE_OPENSSL
+
+using namespace std;
+using namespace facter::util;
+using namespace boost::filesystem;
+
+namespace bs = boost::system;
+namespace lth_file = leatherman::file_util;
+
+using namespace leatherman::util;
+
+namespace facter { namespace facts { namespace windows {
+
+    ssh_resolver::data ssh_resolver::collect_data(collection& facts)
+    {
+        data result;
+        populate_key("ssh_host_rsa_key.pub", 1, result.rsa);
+        populate_key("ssh_host_dsa_key.pub", 2, result.dsa);
+        populate_key("ssh_host_ecdsa_key.pub", 3, result.ecdsa);
+        populate_key("ssh_host_ed25519_key.pub", 4, result.ed25519);
+        return result;
+    }
+
+    void ssh_resolver::populate_key(std::string const& filename, int type, ssh_key& key)
+    {
+        string dataPath;
+        if (!environment::get("programdata", dataPath)) {
+                LOG_DEBUG("error finding programdata: {1}",
+                leatherman::windows::system_error());
+        }
```

Comment:
still have extra spaces here. also, you can fit the contents in one line:
`LOG_DEBUG("error finding programdata: {1}", leatherman::windows::system_error());`